src/pages/DatasetSearch.js

```diff
@@ -194,26 +195,30 @@ export const DatasetSearch = (props) => {
   const version = versions[key] === undefined ? versions['/custom'] : versions[key];
   const isInstitutionQuery = key === 'myinstitution';
 
+  const fullQuery = assembleFullQuery(isSigningOfficial, isInstitutionQuery, version.query);
+  const institutionSet = institutionId === undefined && isInstitutionQuery;
+
   useEffect(() => {
     const init = async () => {
-      if (institutionId === undefined && isInstitutionQuery) {
-        Notifications.showError({ text: 'You must set an institution in your profile to view the `myinstitution` data library' });
-        props.history.push('/profile');
-        return;
-      }
-      try {
-        const query = assembleFullQuery(isSigningOfficial, isInstitutionQuery, version.query);
-        await DataSet.searchDatasetIndex(query).then((datasets) => {
-          setDatasets(datasets);
-          setLoading(false);
-        });
-      } catch (error) {
-        Notifications.showError({ text: 'Failed to load Elasticsearch index' });
+      if (!loaded) {
+        if (institutionSet) {
+          Notifications.showError({ text: 'You must set an institution in your profile to view the `myinstitution` data library' });
+          //props.history.push('/profile');
```

Comment:
```suggestion
          props.history.push('/profile');
```

We should uncomment this line and ensure that `props` is passed in too.